### ISSUE TITLE
Template validation error for evaluated variables

### ISSUE THREAD

**Issue by HahUK** [microsoft/vscode-azurearmtools#380]
The following template causes two template validation errors with ARM Tools v0.8.1. The validation errors are against the template object, not the specific place where the error occurs.

1.) The template variable "default" is not found. We do not reference a variable called default.
2.) If a variable called default is created, a new validation error occurs, "The template variable NSGRules is not valid. The language expression property "WebPorts" can't be evaluated.

Template:
```json
{
    "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
    "contentVersion": "1.0.0.0",
    "parameters": {
        "Environment": {
            "type": "string",
            "metadata": {
                "description": "Name of the Environment"
            }
        }
    },
    "variables": {
        "PRD": {
            "WebPorts": [
                "8510",
                "8520",
                "8530",
                "8540",
                "8550",
                "8560"
            ]
        },
        "TST": {
            "WebPorts": [
                "8010",
                "8020",
                "8030",
                "8040",
                "8050",
                "8060",
                "8510",
                "8520",
                "8530",
                "8540",
                "8550",
                "8560"
            ]
        },
        "NSGRules": {
            "WEBNSGRule": [
                {
                    "name": "[concat('Allow-In-',variables(parameters('Environment')).WebPort[1],'-To-Web')]",
                    "properties": {
                        "access": "Allow",
                        "direction": "Inbound",
                        "priority": 101,
                        "protocol": "*",
                        "sourcePortRange": "*",
                        "sourceAddressPrefix": "VirtualNetwork",
                        "destinationAddressPrefix": "VirtualNetwork",
                        "destinationPortRange": "[variables(parameters('Environment')).WebPort[1]]"
                    }
                }
            ]
        }
    },
    "resources":[]
}
```
I know this is an odd example, we do a lot of dynamic generation of rules based on parameters and this is just a contrived example.

Running:

Test-AzDeployment -TemplateFile .\Test.json -Location uksouth -TemplateParameterObject @{Environment="PRD"} -Verbose

is successful, as is running a real deployment.

Thanks.

**Comment by StephenWeatherford** [comment]
That's pretty weird, but we essentially understand the problem and are looking at solutions.  Thanks for letting us know!

**Comment by StephenWeatherford** [comment]
https://dev.azure.com/devdiv/DevDiv/_boards/board/t/ARM%20Template%20Authoring/Stories/?workitem=1017615

**Comment by StephenWeatherford** [comment]
This was fixed by https://github.com/microsoft/vscode-azurearmtools/pull/383 and was released as a hotfix today as 0.8.2.

**Comment by HahUK** [comment]
Thanks a lot for sorting so quickly.
